test/assert_solidity.go

```diff
@@ -53,29 +54,38 @@ func (assert *Assert) solidityVerification(b backend.ID, vk verifyingKey,
 	out, err := cmd.CombinedOutput()
 	assert.NoError(err, string(out))
 
+	// len(vk.K) - 1 == len(publicWitness) + len(commitments)
+	numOfCommitments := vk.NbPublicWitness() - len(validPublicWitness.Vector().(fr_bn254.Vector))
+
 	// proof to hex
 	var proofStr string
-	var optBackend string
+	var checkerOpts []string
+	checkerOpts = []string{"verify"}
 
 	if b == backend.GROTH16 {
-		optBackend = "--groth16"
+		checkerOpts = append(checkerOpts, "--groth16")
 		var buf bytes.Buffer
 		_proof := proof.(*groth16_bn254.Proof)
 		_, err = _proof.WriteRawTo(&buf)
 		assert.NoError(err)
 		proofBytes := buf.Bytes()
-		// keep only fpSize * 8 bytes; for now solidity contract doesn't handle the commitment part.
-		proofBytes = proofBytes[:32*8]
+		if numOfCommitments == 0 {
+			proofBytes = proofBytes[:32*8] // exclude commitments count if there are no commitments
```

Comment:
I think this is a bit hacky - I think it would be better to have a similar interfaces as for plonk.Proof i.e. to have `Proof.MarshalSolidity` method which encodes only `Ar,Bs,Krs` or `Ar,Bs,Krs,commitments,commitmentsPok`.

Another approachs is that we encode the length of the commitments and use this to instead of having to provide the `--commitment` argument to `gnark-solidity-checker`.

Reply:
I added `MarshalSolidity` for groth16 and simply tests using `interface{ MarshalSolidity() []byte }`